controllers/core/openstackcontrolplane_controller.go

```diff
@@ -160,6 +162,7 @@ func (r *OpenStackControlPlaneReconciler) Reconcile(ctx context.Context, req ctr
 			condition.UnknownCondition(corev1beta1.OpenStackControlPlaneKeystoneAPIReadyCondition, condition.InitReason, corev1beta1.OpenStackControlPlaneKeystoneAPIReadyInitMessage),
 			condition.UnknownCondition(corev1beta1.OpenStackControlPlanePlacementAPIReadyCondition, condition.InitReason, corev1beta1.OpenStackControlPlanePlacementAPIReadyInitMessage),
 			condition.UnknownCondition(corev1beta1.OpenStackControlPlaneGlanceReadyCondition, condition.InitReason, corev1beta1.OpenStackControlPlaneGlanceReadyInitMessage),
+			condition.UnknownCondition(corev1beta1.OpenStackControlPlaneIronicReadyCondition, condition.InitReason, corev1beta1.OpenStackControlPlaneIronicReadyInitMessage),
```

Comment:
I believe this will prevent the openstackcontrolplane CR from reporting Ready if we deploy with Ironic disabled:
https://issues.redhat.com/browse/OSP-21711

Because we initialise the status as Unknown, and then don't run anything in the ReconcileIronic function, so nothing will ever update the status from Unknown. This is my initial proposal for handling disabled services, although I would like to work towards a better implementation that prevents us needing to handle things on a operator by operator basis eventually:

https://github.com/openstack-k8s-operators/openstack-operator/pull/162/commits/aeaf42bb4782aef17a5bc55567bfa1a819e80023

Reply:
Done, thanks